src/pipeline.cc

```diff
@@ -616,13 +616,13 @@ void mainLoop() {
 
   SemaManager manager(
       &project, &wfiles,
-      [&](std::string path, std::vector<Diagnostic> diagnostics) {
+      [&](const std::string &path, std::vector<Diagnostic> diagnostics) {
```

Comment:
`[&]` -> `[]`

Reply:
sema_manager.hh `OnDiagnostic` can be changed to llvm::function_ref now.

Reply:
Lambda capture done. OnDiagnostic & OnDropped part can be delivered by a different PR.